### DISCUSSION TITLE
duplicate 'detection' on all streams

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#194] [Q&A] [answered]
Today all of my cameras mirror the same "car" and "truck" detections as one of the cameras. No matter how many times I star/stop it remains on all cameras. Is there some cache I can delete before starting? (Running the dev branch right now.)

Thank you

**Comment by gniknalu** [comment]
I stopped the container and 'rm'ed it. Then started it with a "docker run -d . ..  .." command. Everything is good now.

**Comment by roflcoopter** [comment, marked as answer]
This is something i have encountered aswell. I suspect its a bug in OpenCVs Darknet implementation that causes it to hang.
I am not sure what to do about this sadly

**Comment by gniknalu** [comment]
This morning I can't seem to get the detection from one camera to stop propagating to all the other cameras. I stop, rm, wait and start and it still happens. (Repeated it 4 times already.)  Is there some cache that can be deleted or tmp directory that can be cleared to fix this bug?

**Reply by roflcoopter** [reply]
Can you check on your GPU with `nvidia-smi`, anything suspicious going on?
I really dont know why this is happening and its impossible to Google it seems :(

**Reply by gniknalu** [reply]
> Can you check on your GPU with `nvidia-smi`, anything suspicious going on?
> I really dont know why this is happening and its impossible to Google it seems :(

When the docker image was stopped, there wasn't anything processing in the GPU (nvidia-smi didn't list anything). But somehow I suspect it might be GPU related. I'll investigate methods of clearing cache for the GPU.

**Reply by roflcoopter** [reply]
It is very hard to investigate issues like this when it cant be reproduced consistently.
My plan was to check if detections seem duplicate and then re-initialize the detector, but if it doesnt help to restart and remove the container that probably wont work.

Just to be clear, you ran `docker stop` and `docker rm` and it did not help?

**Reply by roflcoopter** [reply]
Another question, what driver do you have installed on the host?

**Reply by gniknalu** [reply]
> Just to be clear, you ran `docker stop` and `docker rm` and it did not help?

I have a `systemctl` set up to stop/start it via cron. When it started, I noticed the issue. Used `systemctl` then to stop it and did a `docker rm` then used `systemctl` to start it. Same issue. So, when it was stopped, I tried to investigate the state of the host, docker and GPU but didn't see anything odd. Since nothing was running on the GPU, no python processes were running in docker and didn't see anything in a ps or htop - I did the only thing a person can do. Reboot the host. Then everything worked.

Another note: My goal is to see how long I can go without having to manually interact with anything on the host. So things had been running for about 48 hours (with viseron restarting via cron twice a day) without intervention when the problem happened at 3:00 AM.

**Comment by roflcoopter** [comment]
Could you tell me what hardware you are running on?
I find it strange that you have to restart so often. My instance of Viseron has been running for 11 days without any issues at all

**Reply by gniknalu** [reply]
> Could you tell me what hardware you are running on?
> I find it strange that you have to restart so often. My instance of Viseron has been running for 11 days without any issues at all

It is a slightly older  i7 motherboard with 30GB of RAM and the GPU is a GeForce GTX 1070 Ti. Running Ubuntu 20.04. The drive is a bit small but has plenty of space. (Planning on moving it to an SSD.) 

htop rarely sees any kind of load and the same is true with nvidia-smi

This is some of the output from a `lshw`

```
 *-cpu
      description: CPU
      product: Intel(R) Core(TM) i7-3820 CPU @ 3.60GHz
 *-memory
      description: System Memory
      physical id: 2e
      slot: System board or motherboard
      size: 30GiB
 *-display
      description: VGA compatible controller
      product: GP104 [GeForce GTX 1070 Ti]
      vendor: NVIDIA Corporation
      physical id: 0
      bus info: pci@0000:01:00.0
      version: a1
      width: 64 bits
      clock: 33MHz
 *-disk
      description: ATA Disk
      product: ST6000VN001-2BB1
      physical id: 0.0.0
      bus info: scsi@5:0.0.0
      logical name: /dev/sda
      version: SC60
      serial: ZR104VBQ
      size: 5589GiB (6001GB)
 *-network
      description: Ethernet interface
      product: RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller
      vendor: Realtek Semiconductor Co., Ltd.
```

```
nvidia-smi
Thu Jul  1 14:42:12 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.80       Driver Version: 460.80       CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 107...  Off  | 00000000:01:00.0 Off |                  N/A |
| 37%   61C    P2    46W / 180W |   2233MiB /  8118MiB |     29%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

```

**Reply by gniknalu** [reply]
_I have an idea._ In MQTT you can watch a detection and it will show you the object, label and the rel_x1 rel_y1 rel_x2 rel_y2 etc. I wonder if you can check to see if the same object, label and coordinates are detected in another camera? If so, then ERROR.

Just a thought/suggestion. 

**Thank you again for this fine project!**

**Reply by gniknalu** [reply]
**Further investigation:** After stopping and doing a `docker rm viseron`. I ran a `fuser -v /dev/nvidia*` command to see what is on the GPU. It didn't list anything. So I wonder if it is caching in a file that isn't cleared in the container?
